lib/coin-selection/cardano-wallet-coin-selection.cabal

```diff
@@ -0,0 +1,116 @@
+cabal-version:      2.2
+name:               cardano-wallet-coin-selection
+version:            2022.10.6
+synopsis:           Coin selection algorithms for Cardano Wallet.
+description:        Please see README.md.
+homepage:           https://github.com/input-output-hk/cardano-wallet
+author:             IOHK Engineering Team
+maintainer:         operations@iohk.io
+copyright:          2018-2022 IOHK
+license:            Apache-2.0
+category:           Web
+build-type:         Simple
+
+common language
+  default-language:   Haskell2010
+  default-extensions:
+    NoImplicitPrelude
+    OverloadedStrings
+
+common opts-lib
+  ghc-options: -Wall -Wcompat -fwarn-redundant-constraints
+
+  if flag(release)
+    ghc-options: -O2 -Werror
+
+common opts-exe
+  ghc-options: -threaded -rtsopts -Wall
+
+  if flag(release)
+    ghc-options: -O2 -Werror
+
+flag release
+  description: Enable optimization and `-Werror`
+  default:     False
+  manual:      True
+
+flag scrypt
+  description: Enable compatibility support for legacy wallet passwords.
+  default:     True
```

Comment:
It seems this shouldn't be needed in a coin-selection library.

Reply:
> It seems this shouldn't be needed in a coin-selection library.

I **_think_** you're probably right, that it's not needed.

If it's okay with you, I'll make another PR immediately after this one that fixes this (and also the other suggestions you've made).

Thank-you for spotting this!

Reply:
@Anviking https://github.com/input-output-hk/cardano-wallet/pull/3596